### DISCUSSION TITLE
Fluent UI and NextJS errors

### DISCUSSION THREAD

**Discussion by Giuseppetm** [microsoft/fluentui#24497] [Q&A]
In the last few days I'm trying to convert my project made in `React` to `NextJS`. 
(you can find the branch I'm working on [here](https://github.com/StudentiUniMi/website/tree/feature/next_js/pages)).

I had built it all with Fluent UI and Typescript, and I was hoping there was some decent documentation on how to get it all going but apparently I was daydreaming .. the only doc available is [here](https://github.com/microsoft/fluentui/wiki/Server-side-rendering-and-browserless-testing), and it's basically a javascript file (where is the typescript one??) that doesn't work.

Using `@fluentui/react` package to import components literally breaks the entire application: there are several errors like: 
> :warning: TypeError: Cannot read property 'semanticColors' of undefined
> :warning: TypeError: Cannot read property 'fonts' of undefined
> :warning: TypeError: Cannot read property 'disableGlobalClassNames' of undefined

The only workaround I found was using `office-ui-fabric-react/lib-commonjs/` to import the Fluent UI components, but this thing is not updated and some components do not behave correctly.

A user has made available a typescript version for `_document.tsx` and `_app.tsx` but the problems are the same.
https://github.com/resir014/nextjs-fluentui-quickstart/blob/trunk/pages/_document.tsx

Do you have any suggestions? I would appreciate some help, I don't know what else to try. Thank you!

### Packages version
"react": `17.0.2`
"next": `12.2.5`
"@fluentui/react": `8.30.1`,  (`8.90.1` that is the latest doesn't work either)
"typescript": `4.3.2`

### Errors I encountered using `@fluentui/react` components import
|   |   |
|---|---|
| ![image](https://user-images.githubusercontent.com/52317197/186388856-bb7898ee-75aa-425e-8744-b2fe582e0f03.png)  |  ![image](https://user-images.githubusercontent.com/52317197/186388888-b311cc21-590c-40e9-8018-d77f29dc521b.png) |
|  ![image](https://user-images.githubusercontent.com/52317197/186393577-d0460bf4-b173-4339-b099-b47d63f534e8.png)
 |  |

**Comment by miroslavstastny** [comment]
Unfortunately your branch is not accessible.
The typescript version of `_document.tsx` and `_app.tsx` which you linked is for Fluent UI v7.

We definitely need to improve the documentation, the most up to date state of support (and `_document.tsx`) is described in: https://github.com/microsoft/fluentui/issues/11411#issuecomment-1024158756.

I followed the instructions in [wiki](https://github.com/microsoft/fluentui/wiki/Server-side-rendering-and-browserless-testing#nextjs-setup) and the Next application renders (complaining about id mismatch and icons as mentioned in https://github.com/microsoft/fluentui/issues/11411#issuecomment-1024158756). Could you please provide us with repros (ideally codesandbox or minimal github repo).
